shell/browser/media/media_stream_devices_controller.cc

```diff
@@ -62,17 +68,24 @@ bool MediaStreamDevicesController::TakeAction() {
       request_.video_type ==
           blink::mojom::MediaStreamType::GUM_DESKTOP_VIDEO_CAPTURE) {
     HandleUserMediaRequest();
-    return true;
+    return;
   }
 
+  content::RenderFrameHost* rfh = content::RenderFrameHost::FromID(
+      request_.render_process_id, request_.render_frame_id);
+  if (!rfh)
+    return;
+
+  content::BrowserContext* browser_context = rfh->GetBrowserContext();
+  api::Session* session = api::Session::FromBrowserContext(browser_context);
+  if (session->ChooseMediaDevice(request_, std::move(callback_)))
```

Comment:
1. Sounds OK
2. Sounds like it can be worked around by forward-declaring the `MediaRequestHandler` type? It doesn't need the full type in `api::Session` 🤔 
3. We do this everywhere, not sure there's a better way.  Even Chrome does it between BrowserContext and Profile in some places iirc

Reply:
(2) sure yeah but there's no obvious place to forward-declare it.